contracts/ExampleToken.cdc

```diff
@@ -106,10 +89,74 @@ pub contract ExampleToken: FungibleToken {
                 ExampleToken.totalSupply = ExampleToken.totalSupply - self.balance
             }
         }
+
+        /// The way of getting all the Metadata Views implemented by ExampleToken
+        ///
+        /// @return An array of Types defining the implemented views. This value will be used by
+        ///         developers to know which parameter to pass to the resolveView() method.
+        ///
+        pub fun getViews(): [Type]{
+            return [Type<FungibleTokenMetadataViews.FTView>(),
+                    Type<FungibleTokenMetadataViews.FTDisplay>(),
+                    Type<FungibleTokenMetadataViews.FTVaultData>()]
+        }
+
+        /// The way of getting a Metadata View out of the ExampleToken
+        ///
+        /// @param view: The Type of the desired view.
+        /// @return A structure representing the requested view.
+        ///
+        pub fun resolveView(_ view: Type): AnyStruct? {
+            switch view {
+                case Type<FungibleTokenMetadataViews.FTView>():
+                    return FungibleTokenMetadataViews.FTView(
+                        ftDisplay: ExampleToken.returnDisplayView(),
+                        ftVaultData: ExampleToken.returnVaultDataView()
+                    )
+                case Type<FungibleTokenMetadataViews.FTDisplay>():
+                    return ExampleToken.returnDisplayView()
+                case Type<FungibleTokenMetadataViews.FTVaultData>():
+                    return ExampleToken.returnVaultDataView()
+            }
+            return nil
+        }
+    }
+
+    // Auxiliary function to return the FTVaultData view
+    access(contract) fun returnVaultDataView(): FungibleTokenMetadataViews.FTVaultData {
```

Comment:
Why do you want the view resolvers to be outside the Vault type? I don't necessarily think it is bad, I'm just curious to hear your reasoning. I would probably prefer to have them defined in the `Vault` type

Reply:
Just because when you return the `FTView` you also need to create a new `FTVaultData` and a new `FTDisplay` I kind of hate to see duplicate code

Reply:
Yep fixed, I'm kind of embarrassed I didn't think about that 🤦